File: sams-backend/src/main/java/com/sams/entity/Printer.java
package com.sams.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "m_printer", schema = "base")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Printer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "printer_id")
    private Long printerId;

    @Column(name = "org_id")
    private Long orgId;

    @Column(name = "location_id")
    private Long locationId;

    @Column(name = "printer_name")
    private String printerName;

    @Column(name = "printer_model_id")
    private Long printerModelId;

    @Column(name = "printer_model")
    private String printerModel;

    @Column(name = "printer_manufacturer")
    private String printerManufacturer;

    @Column(name = "communication_type")
    private String communicationType;

    @Column(name = "default_printer")
    private Boolean defaultPrinter;

    @Column(name = "active")
    private Boolean active;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "created_dt")
    private LocalDateTime createdDt;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_dt")
    private LocalDateTime updatedDt;

}

File: SAMSUI/src/app/Model/base/printer.ts

export class PrinterModel {

    public printerId?: number;
    public orgId?: number;
    public locationId?: number;
    public locationName?: string;

    public printerName?: string;
    public printerManufacturer?: string;
    public printerModelId?: number;
    public printerModel?: string;
    public communicationType?: string;
    
    public active?: boolean; 
    public columnName ?:string;
    public direction ?:string;
    
    public createdBy?: string;
    public createdDt?: Date;
    public updatedBy?: string;
    public updatedDt?: Date;

    public pageNumber?: number;
    public recordsPerPage?: number;
    public updatedDtDisp?: string;
    public createdDtDisp?: string;
    public logInUserOrgId?: number;
    public logInUserLocId?: number;
    public logInUserId?: number;  
    public searchValue?: string;
    public searchValue1?: string;

}